logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -37,5 +37,231 @@ def take
     def poll(millis)
       @queue.poll(millis, TimeUnit::MILLISECONDS)
     end
+
+    def write_client
+      WriteClient.new(self)
+    end
+
+    def read_client
+      ReadClient.new(self)
+    end
+
+    class ReadClient
+      # We generally only want one thread at a time able to access pop/take/poll operations
+      # from this queue. We also depend on this to be able to block consumers while we snapshot
+      # in-flight buffers
+
+      def initialize(queue)
+        @queue = queue
+        @mutex = Mutex.new
+        @inflight_batches = {}
+        @batch_size = 125
+        @wait_for = 5
+      end
+
+      def set_batch_details(batch_size, wait_for)
```

Comment:
I find it a bit weird that this is not set in the constructor. this is not something that will or should change in the life of that object? I think it would make more sense to bubble up these settings from the `start_worker` to the Pipeline constructor where the `ReadClient` is instantiated.